UPGRADE.md

```diff
@@ -0,0 +1,90 @@
+Note about upgrading: Doctrine uses static and runtime mechanisms to raise
+awareness about deprecated code.
+
+- Use of `@deprecated` docblock that is detected by IDEs (like PHPStorm) or
+  Static Analysis tools (like Psalm, phpstan)
+- Use of our low-overhead runtime deprecation API, details:
+  https://github.com/doctrine/deprecations/
+
+# Upgrade to 3.0
+
+## BC Break: removed `ObjectManager::merge()`
+
+`ObjectManagerDecorator::merge()` is removed as well.
+
+## BC Break: removed support for `doctrine/cache`
+
+Removed support for using doctrine/cache for metadata caching. The
+`setCacheDriver` and `getCacheDriver` methods have been removed from
+`Doctrine\Persistence\Mapping\AbstractMetadata`. Please use `getCache` and
+`setCache` with a PSR-6 implementation instead.
+
+## BC Break: changed signatures
+
+`$objectName` has been dropped from the signature of `ObjectManager::clear()`.
+
+```diff
+- public function clear($objectName = null)
++ public function clear(): void
+```
+
+Also, native parameter type declarations have been added on all public APIs.
+Native return type declarations have not been added so that it is possible to
+implement types compatible with both 2.x and 3.x.
+
+## BC Break: Removed `PersistentObject`
+
+Please implement this functionality directly in your application if you want
+ActiveRecord style functionality.
+
+# Upgrade to 2.4
+
+## Deprecated `AnnotationDriver`
+
+Since attributes were introduced in PHP 8.0, annotations are deprecated. Use
+`ColocatedMappingDriver` instead. This will involve implementing
```

Comment:
I should have asked in the PR but here I am now - is it right to advertise a trait here without further explaination?